### PR TITLE
Fix concurrency issues while checking Raft sessions [5.0.z]

### PR DESCRIPTION
**Pull request by arodionov**
In a high-load environment, periodically running the `CheckInactiveSessions` task in `RaftSessionService` can cause concurrency issues if the list of sessions changes at the same time.

Backport of: https://github.com/hazelcast/hazelcast/pull/21334, https://github.com/hazelcast/hazelcast/pull/21266

### CODE REVIEW

**Comment by arodionov** [comment]
run-lab-run

**Comment by arodionov** [comment]
run-ee-compile

File: hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/blocking/BlockingResource.java [hunk 1]
```diff
@@ -161,12 +174,14 @@ protected final void cloneForSnapshot(BlockingResource<W> clone) {
 
     @Override
     public void writeData(ObjectDataOutput out) throws IOException {
-        out.writeObject(groupId);
-        out.writeString(name);
-        out.writeInt(waitKeys.size());
-        for (Entry<Object, WaitKeyContainer<W>> e : waitKeys.entrySet()) {
-            out.writeObject(e.getKey());
-            out.writeObject(e.getValue());
+            out.writeObject(groupId);
```
Comment (frant-hartm) [inline hazelcast/src/main/java/com/hazelcast/cp/internal/datastructures/spi/blocking/BlockingResource.java:177]: The formatting here is incorrect.

**Review by frant-hartm — APPROVED** [review]
LGTM.

You should find 1 more reviewer.

**Pull request merged**